editor/src/clj/editor/gui.clj

```diff
@@ -843,83 +843,150 @@
-        current-layout (:current-layout trivial-gui-scene-info)
-        changes (if (nil? changes-fn)
-                  {prop-kw new-value}
-                  (changes-fn evaluation-context node-id prop-kw old-value new-value))]
-    (if (str/blank? current-layout)
-      (into []
-            (mapcat (fn [[prop-kw value]]
-                      (if (nil? value)
-                        (g/clear-property node-id prop-kw)
-                        (g/set-property node-id prop-kw value))))
-            changes)
-      (g/update-property
-        node-id :layout->prop->override
-        update current-layout
-        (fn [prop->override]
+(defn- layout-property-changes-tx-data
+  "Takes changes returned by a changes-fn associated with a layout property
+  and returns the transaction steps that perform those changes to the specified
+  layout in the GuiNode."
+  [layout-name node-id changes]
+  {:pre [(string? layout-name)]}
+  (if (str/blank? layout-name)
+    (coll/transfer changes []
+      (mapcat (fn [[prop-kw value]]
+                (if (nil? value)
+                  (g/clear-property node-id prop-kw)
+                  (g/set-property node-id prop-kw value)))))
+    (g/update-property
+      node-id :layout->prop->override
+      update layout-name
+      (fn [prop->override]
+        (coll/not-empty
           (reduce-kv (fn [prop->override prop-kw new-value]
                        (if (nil? new-value)
                          (dissoc prop->override prop-kw)
                          (assoc prop->override prop-kw new-value)))
                      prop->override
                      changes))))))
 
-(defn basic-layout-property-edit-type-clear-fn [node-id prop-kw]
+(defn- layout-property-clears-tx-data
+  "Takes a sequence of prop-kws, possibly based on the return value of a
+  changes-fn, and returns the transaction steps that will clear overrides for
+  those properties."
+  [layout-name node-id cleared-prop-kws]
+  (when (coll/not-empty cleared-prop-kws)
+    (if (str/blank? layout-name)
+      (coll/mapcat
+        #(g/clear-property node-id %)
+        cleared-prop-kws)
+      (g/update-property
+        node-id :layout->prop->override
+        update layout-name
+        (fn [prop->override]
+          (coll/not-empty
+            (apply dissoc prop->override cleared-prop-kws)))))))
+
+(defn- prop->value-for-specific-layout
+  [node-id layout-name evaluation-context]
+  (let [layout->prop->value (g/node-value node-id :layout->prop->value evaluation-context)
+        prop->value (get layout->prop->value layout-name ::not-found)]
+    (when (= ::not-found prop->value)
+      (throw (ex-info (format "Layout '%s' does not exist in the scene." layout-name)
+                      {:node-id node-id
+                       :layout-name layout-name
+                       :layout-name-candidates (vec (sort (keys layout->prop->value)))})))
+    prop->value))
```

Comment:
The map returned here should contain all values, not only overrides, so it could be used to query the value of a property in a specific layout.